pxtwinrt/serial.ts

```diff
@@ -56,13 +56,25 @@ namespace pxt.winrt {
         port.device.baudRate = 115200;
         let stream = port.device.inputStream;
         let reader = new Windows.Storage.Streams.DataReader(stream);
+        let serialBuffers: {[source: string]: string} = {};
+        const maxBufferLength = 255;
         let readMore = () => reader.loadAsync(32).done((bytesRead) => {
             let msg = reader.readString(Math.floor(bytesRead / 4) * 4);
-            window.postMessage({
-                type: 'serial',
-                data: msg,
-                id: id
-            }, "*");
+            const data = msg || ""
+            const source = id || "?"
+            for (let i = 0; i < data.length; ++i) {
+                const char = data[i]
+                serialBuffers[source] = serialBuffers[source] ? serialBuffers[source] + char : char
+                if (/.*\n/.test(serialBuffers[source]) || serialBuffers[source].length > maxBufferLength) {
```

Comment:
What is the intent of this regex?

Reply:
We want to post a chunk of serial data when either (a) we hit a newline of (b) the buffer is full.  This regex tests for case (a).

Reply:
Since you are adding the last character, ``char`` is your last letter in the buffer.

Reply:
Oh yeah ... fixed :D